buildkite/bazelci.py

```diff
@@ -1285,6 +1304,54 @@ def execute_bazel_build(
         handle_bazel_failure(e, "build")
 
 
+def expand_test_target_patterns(bazel_binary, platform, test_targets):
+    if not any(t for t in test_targets if t.endswith(":all") or t.endswith("...")):
+        return test_targets
+
+    eprint("Resolving test targets via bazel query")
+    output = execute_command_and_get_output(
+        [bazel_binary]
+        + common_startup_flags(platform)
+        + [
+            "--nomaster_bazelrc",
+            "--bazelrc=/dev/null",
+            "query",
+            "tests(set({}))".format(" ".join("'{}'".format(t) for t in test_targets if t != "--")),
+        ],
+        print_output=False,
+    )
+    return [t for t in output.split("\n") if t.startswith("//")]
+
+
+def get_targets_for_shard(build_targets, test_targets, shard_id, shard_count):
+    # TODO(fweikert): implement a more sophisticated algorithm
+    all_targets = [("build", t) for t in sorted(build_targets)] + [
```

Comment:
We can simplify this entire function using the "step" argument of Python slices into two lines :D

```python
build_targets_for_this_shard = sorted(build_targets)[shard_id::shard_count]
test_targets_for_this_shard = sorted(test_targets)[shard_id::shard_count]
```

Reply:
Yeah, I forgot about the step argument -.-

However, we need to adjust the starting index for test_targets to get the same outcome.